static/app/components/workflowEngine/form/environmentSelector.tsx

```diff
@@ -0,0 +1,70 @@
+import {useMemo} from 'react';
+
+import type {SelectOption} from 'sentry/components/core/compactSelect';
+import {CompactSelect} from 'sentry/components/core/compactSelect';
+import {t} from 'sentry/locale';
+import useProjects from 'sentry/utils/useProjects';
+
+interface EnvironmentSelectorProps {
+  onChange: (value: string) => void;
+  value: string;
+}
+
+export function EnvironmentSelector({value, onChange}: EnvironmentSelectorProps) {
+  const {projects, initiallyLoaded: projectsLoaded} = useProjects();
+
+  const userProjectEnvironments = useMemo<Set<string>>(() => {
+    return new Set(
+      projects.flatMap(project => {
+        if (project.isMember) {
+          return project.environments;
+        }
+        return [];
+      })
+    );
+  }, [projects]);
+
+  const otherEnvironments = useMemo<Set<string>>(() => {
+    return new Set(
+      projects.flatMap(project => {
+        if (!project.isMember) {
+          return project.environments;
+        }
+        return [];
+      })
+    );
+  }, [projects]).difference(userProjectEnvironments);
+
+  const options = [
+    {
+      key: 'my-projects',
+      label: t('Environments in My Projects'),
+      options: setToOptions(userProjectEnvironments),
+    },
+    {
+      key: 'other-projects',
+      label: t('Other Environments'),
+      options: setToOptions(otherEnvironments),
+    },
+  ];
```

Comment:
this whole thing should just be one big useMemo that returns the options

Reply:
good point, fixed in https://github.com/getsentry/sentry/pull/92077/commits/d09f8f467d7f5c6bcdb8c8cc443727c14da1e3ff